app/ui-react/syndesis/src/modules/data/shared/VirtualizationUtils.ts

```diff
@@ -194,3 +204,67 @@ export function isDvConnectionSelected(conn: Connection) {
   }
   return isSelected;
 }
+
+/**
+ * Generate preview SQL for the specified view definition
+ * @param viewDefinition the ViewDefinition
+ */
+export function getPreviewSql(viewDefinition: ViewDefinition): string {
+  if (viewDefinition) {
+    // TODO: This assumes a single source view.  Will need to expand capability later
+    const sourcePath = viewDefinition.sourcePaths[0];
+    if (sourcePath) {
+      const tableName = getPreviewTableName(sourcePath);
+      return 'SELECT * FROM ' + tableName + ';';
+    }
+  }
+  return '';
+}
+
+/**
+ * Generates the table name for the preview query, given the source path.
+ * Example sourcePath: (connection=pgConn/schema=public/table=account)
+ * @param sourcePath the path for the view source
+ */
+function getPreviewTableName(sourcePath: string): string {
+  const sourceNodeName = getSourceName(sourcePath);
+  const connectionName = getConnectionName(sourcePath);
+
+  // Assemble the name, utilizing the schema model suffix
+  return (
+    connectionName.toLowerCase() + SCHEMA_MODEL_SUFFIX + '.' + sourceNodeName
+  );
+}
+
+/**
+ * Get the connection name from the supplied source path
+ * Example sourcePath: (connection=pgConn/schema=public/table=account)
+ * @param sourcePath the view definition sourcePath
+ */
+function getConnectionName(sourcePath: string): string {
+  const fqnArray = sourcePath.split('/', 10);
+
+  // Connection name is contained in the first segment
+  const connectionSegment = fqnArray[0];
+  const parts = connectionSegment.split('=', 2);
+  const connName = parts[1];
+
+  return connName;
+}
+
+/**
+ * Get the source name from the supplied source path
+ * Example sourcePath: (connection=pgConn/schema=public/table=account)
+ * @param sourcePath the view definition sourcePath
+ */
+function getSourceName(sourcePath: string): string {
+  const fqnArray = sourcePath.split('/', 10);
+  const arrayLength = fqnArray.length;
+
+  // Source name is contained in the last segment
+  const nodeSeqment = fqnArray[arrayLength - 1];
+  const parts = nodeSeqment.split('=', 2);
+  const nodeName = parts[1];
+
+  return nodeName;
```

Comment:
Since this and the previous method share the logic, you could write a helper function. Two options, the first assumes a fixed ordering:

```ts
function parseSourcePath(source: string): [string, string, string] {
  return source
    .replace(/[\(\)]/gi, '') // remove the wrapping ()
    .split('/')
    .map(p => 
      p.split('=')[1] // returns only the value part, assuming a fixed ordering
    );
}
// usage
const [connection, schema, table] = parseSourcePath('(connection=pgConn/schema=public/table=account)');
```

Or, if you can't assume a fixed ordering:

```ts
function parseSourcePath(source: string): {connection: string, schema: string, table: string} {
  return source
    .replace(/[\(\)]/gi, '') // remove the wrapping ()
    .split('/')
    .map(p => p.split('=')) // convert the string 'k=v' to a tuple [k, v]
    .reduce((acc, [k, v]) => ({ ...acc, [k]: v}), {}); // aggregate the tuples in a single object
}
// usage
const {connection, schema, table} = parseSourcePath('(connection=pgConn/schema=public/table=account)');
```

Reply:
The connection is always the first segment, but the depth of the remaining segments and segment names can vary depending on the connection type.  I think I can improve based on your suggestions though - I'll look at it.

Reply:
pushed rework of the functions to simplify